File: TwoPointer/9.palindrome-number.go
package leetcode

// @leet start
func isPalindrome(x int) bool {
	var (
		num = x
		res = 0
	)
	for num > 0 {
		res = res*10 + num%10
		num /= 10
	}
	return x == res
}

// @leet end

File: TwoPointer/9.palindrome-number.py
# @leet start
class Solution:
    def isPalindrome(self, x: int) -> bool:
        num = x
        res = 0
        while num > 0:
            res = res * 10 + num % 10
            num //= 10
        return res == x


# @leet end

